### PR TITLE
Quality of Life refactors

### PR DESCRIPTION
**Pull request by shirsho-12**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/edit/EditRecurringExpenseManagerCommand.java
Comment (nicleejy) [inline src/main/java/seedu/address/logic/commands/edit/EditRecurringExpenseManagerCommand.java:1]: A good decision to let `EditCommand` be an interface which extends from `Command`, follows OOP principles more closely.

File: src/main/java/seedu/address/model/category/UniqueCategoryList.java [hunk 1]
```diff
@@ -43,12 +43,12 @@ public void add(Category newCategory) {
...
+     * @param category Category to remove
      */
-    public void remove(Category toRemove) {
-        requireNonNull(toRemove);
+    public void remove(Category category) {
+        requireNonNull(category);
 
-        if (!internalListOfCategories.remove(toRemove)) {
+        if (!internalListOfCategories.remove(category)) {
             //Throw an exception here later
```
Comment (nicleejy) [inline src/main/java/seedu/address/model/category/UniqueCategoryList.java:52]: This part should be implemented by throwing an exception if the category to remove not being present in the list. If the Category was already checked to have been contained earlier, then this need not be an if statement.

File: src/main/java/seedu/address/model/expense/RecurringExpenseManager.java [hunk 2]
```diff
@@ -209,31 +197,24 @@ public boolean equals(Object object) {
...
             return false;
         }
-
         if (!recurringExpense.expenseCategory.equals(this.expenseCategory)) {
             return false;
         }
-
         if (!recurringExpense.expenseName.equals(this.expenseName)) {
             return false;
         }
```
Comment (nicleejy) [inline src/main/java/seedu/address/model/expense/RecurringExpenseManager.java:214]: There could be a less repetitive way of doing this, can consider refactoring, but it is not a big issue.

**Review by nicleejy — APPROVED** [review]
Clear refactoring of the codebase to follow OOP principles. Looks good to me overall!

File: src/main/java/seedu/address/model/expense/RecurringExpenseManager.java (hunk 2 above)
Comment (shirsho-12) [reply src/main/java/seedu/address/model/expense/RecurringExpenseManager.java:214]: i agree, i'll refactor it

**Pull request merged**